extensions/tls-registry/cli/src/main/java/io/quarkus/tls/cli/GenerateCertificateCommand.java

```diff
@@ -0,0 +1,168 @@
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+
+import io.smallrye.common.os.OS;
+import me.escoffier.certs.CertificateGenerator;
+import me.escoffier.certs.CertificateRequest;
+import me.escoffier.certs.Format;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "generate-certificate", mixinStandardHelpOptions = true, description = "Generate a TLS certificate with the Quarkus Dev CA if available.")
+public class GenerateCertificateCommand implements Callable<Integer> {
+
+    @CommandLine.Option(names = { "-n",
+            "--name" }, description = "Name of the certificate. It will be used as file name and alias in the keystore", required = true)
+    String name;
+
+    @CommandLine.Option(names = { "-p",
+            "--password" }, description = "The password of the keystore. Default is 'password'", defaultValue = "password", required = false)
+    String password;
+
+    @CommandLine.Option(names = { "-c",
+            "--cn" }, description = "The common name of the certificate. Default is 'localhost'", defaultValue = "localhost", required = false)
+    String cn;
+
+    @CommandLine.Option(names = { "-d",
+            "--directory" }, description = "The directory in which the certificates will be created. Default is `.certs`", defaultValue = ".certs")
+    String directory;
+
+    @CommandLine.Option(names = { "-r",
+            "--renew" }, description = "Whether existing certificates will need to be replaced", defaultValue = "false")
+    boolean renew;
+
+    @CommandLine.Option(names = {
+            "--self-signed" }, description = "Generate a self-signed certificate", defaultValue = "false", hidden = true)
+    boolean selfSigned;
+
+    static {
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+    }
+
+    static System.Logger LOGGER = System.getLogger("generate-quarkus-ca");
+
+    @Override
+    public Integer call() throws Exception {
+        LOGGER.log(INFO, "\uD83D\uDD0E Looking for the Quarkus Dev CA certificate...");
+
+        if (!CA_FILE.exists() || !PK_FILE.exists() || selfSigned) {
+            LOGGER.log(INFO, "\uD83C\uDFB2 Quarkus Dev CA certificate not found. Generating a self-signed certificate...");
+            generateSelfSignedCertificate();
+            return 0;
+        }
+
+        LOGGER.log(INFO, "\uD83D\uDCDC Quarkus Dev CA certificate found at {0}", CA_FILE.getAbsolutePath());
+        X509Certificate caCert = loadRootCertificate(CA_FILE);
+        PrivateKey caPrivateKey = loadPrivateKey();
+
+        createSignedCertificate(caCert, caPrivateKey);
+
+        LOGGER.log(INFO, "🔥 Signed Certificate generated successfully and exported into `{0}-keystore.p12`", name);
```

Comment:
Should we try to be more consistent with the feedback we have elsewhere?

I'm not sure a fire emoji is a good way to say something is successful.

We had the discussion with Roberto here: https://github.com/quarkusio/quarkus/pull/41203#issuecomment-2196726872

Reply:
✅ it is!